### PR TITLE
test: integration test for connector list before manually setting up any connector

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Integration tests: List connectors after initializing a new Logto instance and before manually setting up any connector.

There should be no connectors, or all default connectors should be disabled.

## Testing

Ditto

### CODE REVIEW

File: packages/integration-tests/tests/connectors.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { ConnectorDto } from '@logto/schemas';
+
+import { getConnectorDetails, listConnectors } from '@/connector-api';
+
+describe('list connectors and get details before setting up', () => {
```
Comment (gao-sun) [inline packages/integration-tests/tests/connectors.test.ts:5]: what is setting up? it's ambigious

File: packages/integration-tests/tests/connectors.test.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  it('should succeed to get all connectors and they are all disabled', async () => {
```
Comment (gao-sun) [inline packages/integration-tests/tests/connectors.test.ts:6]: succeed to get? correct grammar?

File: packages/integration-tests/tests/connectors.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    expect(() => {
+      for (const connectorDto of JSON.parse(response.body) as ConnectorDto[]) {
+        expect(connectorDto.enabled).toBeFalsy();
+        expect(connectorDto.config).toEqual({});
+      }
+    }).not.toThrow();
+  });
+
+  it('should succeed to get connector details', async () => {
+    const response = await getConnectorDetails('facebook-universal');
```
Comment (gao-sun) [inline packages/integration-tests/tests/connectors.test.ts:18]: don't assume what connector will be loaded

File: packages/integration-tests/src/connector-api.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { authedAdminApi } from '@/api';
+
+export const listConnectors = async () => authedAdminApi.get('connectors');
```
Comment (xiaoyijun) [inline packages/integration-tests/src/connector-api.ts:3]:
```suggestion
export const listConnectors = async () => authedAdminApi.get('connectors').json< ConnectorDto[]>();
```

File: packages/integration-tests/src/connector-api.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+export const getConnectorDetails = async (connectorId: string) =>
+  authedAdminApi.get(`connectors/${connectorId}`);
```
Comment (xiaoyijun) [inline packages/integration-tests/src/connector-api.ts:6]:
```suggestion
  authedAdminApi.get(`connectors/${connectorId}`).json< ConnectorDto>();
```

File: packages/integration-tests/tests/connectors.test.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+describe('list connectors and get details before setting up', () => {
+  it('should succeed to get all connectors and they are all disabled', async () => {
+    const response = await listConnectors();
+    expect(response).toHaveProperty('statusCode', 200);
+    expect(() => {
+      for (const connectorDto of JSON.parse(response.body) as ConnectorDto[]) {
+        expect(connectorDto.enabled).toBeFalsy();
+        expect(connectorDto.config).toEqual({});
+      }
+    }).not.toThrow();
```
Comment (xiaoyijun) [inline packages/integration-tests/tests/connectors.test.ts:14]: This logic can be improved once we set the response type like the comments above.

File: packages/integration-tests/tests/connectors.test.ts (hunk 3 above)
Comment (IceHe) [reply packages/integration-tests/tests/connectors.test.ts:18]: removed

File: packages/integration-tests/tests/connectors.test.ts (hunk 2 above)
Comment (IceHe) [reply packages/integration-tests/tests/connectors.test.ts:6]: rephrased.

File: packages/integration-tests/tests/connectors.test.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { listConnectors } from '@/connector-api';
+
+test('connector flow', async () => {
+  // List connectors after initializing a new Logto instance
+  const connectorDtos = await listConnectors();
+
+  // There should be no connectors, or all connectors should be disabled.
+  for (const connectorDto of connectorDtos) {
+    expect(connectorDto.enabled).toBeFalsy();
+  }
```
Comment (gao-sun) [inline packages/integration-tests/tests/connectors.test.ts:10]: describe the case? or it's a part of a bigger flow?

**Review by gao-sun — APPROVED** [review]

**Pull request merged**